src/time/Waveform.cpp

```diff
@@ -43,9 +44,11 @@ Eigen::VectorXd Waveform::sample(double normalizedDt) const
 
   PRECICE_ASSERT(usedDegree >= 1);
 
-  const auto data = _timeStepsStorage.getTimesAndValues();
+  const auto data   = _timeStepsStorage.getTimesAndValues();
+  auto       sample = _bspline.interpolateAt(data.first, data.second, usedDegree, normalizedDt, _recomputeBuffer);
+  _recomputeBuffer  = false;
```

Comment:
If `_recomputeBuffer` is inside `Bspline`, you can also just take care of setting it `false` inside `Bspline::computeMatrices()`